kornia/contrib/kmeans.py

```diff
@@ -0,0 +1,155 @@
+# based on https://github.com/subhadarship/kmeans_pytorch
+
+import torch
+
+
+class KMeans:
+    def __init__(
+        self,
+        num_clusters: int,
+        cluster_centers: torch.Tensor,
+        tolerance: float = 10e-4,
+        max_iterations: int = 0,
+        device: torch.device = torch.device('cpu'),
```

Comment:
add `from __future__ import annotations`

```suggestion
        device: torch.device | None = None,
```

Reply:
I don’t think device should be needed here. We can inherit from clusters centers

Reply:
And possibly should be a nn parameter so that can get benefited of the semantics

Reply:
agree with edgar (+1) 😄